pkgs/build-support/rust/lib/default.nix

```
@@ -12,17 +12,25 @@ rec {
   # hostPlatform-targeted compiler -- for example, `-m64` being
   # passed on a build=x86_64/host=aarch64 compilation.
   envVars = let
+
+    isAarch64Static = platform: platform.isAarch64 && platform.isMusl && platform.isStatic;
+
     ccForBuild = "${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc";
     cxxForBuild = "${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}c++";
+    linkerForBuild = ccForBuild;
+
     ccForHost = "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
     cxxForHost = "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}c++";
+    linkerForHost = if isAarch64Static stdenv.targetPlatform then "${buildPackages.lld}/bin/ld.lld" else ccForHost;
```

We should check whether stdenv's linker is lld already, so we don't unnecessarily override it with a potentially different version.  Something like

```suggestion
    linkerForHost = if isAarch64Static stdenv.targetPlatform && !stdenv.cc.bintools.isLLVM then "${buildPackages.lld}/bin/ld.lld" else ccForHost;
```

---

(And also for the target.)